### ISSUE TITLE
Any chance for an API?

### ISSUE THREAD

**Issue by MMoMM-org** [Fevol/obsidian-translate#37]
I would like to pass a word, source language, target language to the api and get back the translated word.

Probably the return value needs to be an array with the translation word and perhaps the descriptions of it.

Would that be possible?

**Comment by MMoMM-org** [comment]
Hi, thanks that helps a lot.
And I noticed that I was tackling it wrong, I need to figure out how to get into a dictionary for my specific ask as I want to translate just one word... which can have several meanings.

You plugin is more for full sentences.

But nevertheless:

1) the code above helped a lot
2) would be cool if you could in the future write an api which wraps around the different implementations and handles errors in itself
3) as you probably close this issue I would suggest to get this information into the documentation, could be helpful to others.

And thanks again for writing this up, this helped a lot.

**Comment by Fevol** [comment]
> I need to figure out how to get into a dictionary for my specific ask as I want to translate just one word... which can have several meanings. (...) You plugin is more for full sentences.

That's true, I'll be trying to get some sort of dictionary integration going for the future which should give better translations for single words/terms, but as mentioned before, that's going to be a little ways off.

---

> 2. would be cool if you could in the future write an api which wraps around the different implementations and handles errors in itself

The API does handle all internal errors, but when you make a request, it isn't guaranteed that you get back a translation (due to service being offline, invalid authentication info, ...). So that's why I'm checking `status_code === 200` in the example I gave; if any error had occurred, then you'd get a status code ≠ 200, and an (error) message will be provided in the return values.

The reason why the message isn't automatically displayed, is so that people can decide what to do with it: completely ignoring it, throwing an error, putting it in console log, making a notice in Obsidian, etc..

Basically what I'm saying is, it's perfectly fine to just use the code below, but you will not know if/how the translation failed.
```ts
// If an error occurs during translation, translated_text = undefined
const translated_text = await translator.translate(TEXT, FROM, TO, APPLY_GLOSSARY).translation
```

---

> 3. as you probably close this issue I would suggest to get this information into the documentation, could be helpful to others.

Don't worry, I won't be closing the issue till we're both satisfied! I've added a brief API documentation [here](https://github.com/Fevol/obsidian-translate/blob/master/docs/API.md), for everyone that needs it. If you have any suggestions on how I could improve it, please let me know!
